### DISCUSSION TITLE
Does cue have anything like godoc?

### DISCUSSION THREAD

**Discussion by bbkane** [cuelang/cue#732] [Ideas]
I'm trying to use cue for application modeling with hof and trying to understand files like https://github.com/hofstadter-io/hof/blob/_dev/schema/datamodel.cue . I don't understand what some of the fields do there and I thought how nice it would be if it was possible to add annotations that could generate HTML like godoc does for go code. I know cue has [attributes](https://cuelang.org/docs/references/spec/#attributes) - could those be used to generate docs?

**Comment by myitcv** [comment]
Related to https://github.com/cuelang/cue/discussions/730 in so far as you're referring to a dependency. 

`cue doc` (the `go doc` equivalent), which would form the basis of something like https://pkg.go.dev/, should also exist. That really falls under https://github.com/cuelang/cue/issues/337

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/732.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
